### PR TITLE
checkURL interface

### PR DESCRIPTION
**Pull request by alexcos20**

### CODE REVIEW

**Review by calina-c — APPROVED** [review]
nice :D

File: src/provider/Provider.ts [hunk 1]
```diff
@@ -11,6 +11,11 @@ import { DDO } from '../ddo/DDO'
 
 const apiPath = '/api/v1/services'
 
+export interface urlDetails {
```
Comment (kremalicious) [inline src/provider/Provider.ts:14]: types and interfaces should always start uppercase

**Comment by kremalicious** [comment]
So is it possible to get that method without initializing the whole `ocean` object? Otherwise we would need to block adding any files in UI as long as user does not have any wallet connected

File: src/provider/Provider.ts [hunk 2]
```diff
@@ -73,20 +78,23 @@ export class Provider extends Instantiable {
     }
   }
 
-  public async checkURL(url: string): Promise<Record<string, string>> {
```
Comment (kremalicious) [inline src/provider/Provider.ts:76]: should be named `checkUrl`

**Comment by kremalicious** [comment]
not sure yet why https://github.com/oceanprotocol/provider/pull/57 introduces `/checkUrl` &  `/assetInfo` and what is best to use here

File: src/provider/Provider.ts (hunk 1 above)
Comment (alexcos20) [reply src/provider/Provider.ts:14]: it was indented to mimic https://github.com/oceanprotocol/market/blob/main/src/utils/index.ts#L45
I'm going to update it

**Comment by alexcos20** [comment]
> So is it possible to get that method without initializing the whole `ocean` object? Otherwise we would need to block adding any files in UI as long as user does not have any wallet connected

You need a provider class.  That should be possible without web3

**Comment by alexcos20** [comment]
> not sure yet why [oceanprotocol/provider#57](https://github.com/oceanprotocol/provider/pull/57) introduces `/checkUrl` & `/assetInfo` and what is best to use here

checkUrl takes an URL (http:///www.example.com) and check if your selected provider is able to access it and gets file information (length & type)
/assetInfo takes a DID with some files (you don't have the URLs) and will send back the same interface, but with updated values.

Let's say that you published www.example.com as DID1.  and you are updating the content, but in the DDO you still have the old length of the file.   By calling /assetInfo, provider can decrypt the URLs , check if they are still valid, get the file info and then you can update your DDO with the new info

**Comment by alexcos20** [comment]
superseded by #544

**Pull request closed without merging**